### PR TITLE
implement cpu model and mode selection

### PR DESCRIPTION
**Pull request by mmoll**
closes GH-18

### CODE REVIEW

**Comment by mmoll** [comment]
The most common use case is to set `:cpu=>{:mode=>"host-model"}`.

I tried to keep this extendable if further cpu parameters will get added, but would state the current ERB approach gave not the best result.

**Comment by mmoll** [comment]
anybody? :)

File: lib/fog/libvirt/models/compute/templates/server.xml.erb [hunk 1]
```diff
@@ -13,6 +13,23 @@
     <apic/>
     <pae/>
   </features>
+<% if !cpu.empty? -%>
+<% if cpu[:mode] -%>
```
Comment (plribeiro3000) [inline lib/fog/libvirt/models/compute/templates/server.xml.erb]: Do you need to check for both emptiness and the `mode` key?

Comment (mmoll) [inline lib/fog/libvirt/models/compute/templates/server.xml.erb]: The situation is a bit messy (https://libvirt.org/formatdomain.html#elementsCPU) and for the sake for future-proveness (in theory more parameters could be supported) and setting default values I did it that way.

Comment (plribeiro3000) [inline lib/fog/libvirt/models/compute/templates/server.xml.erb]: Hmmm. I guess you are right here. =)

**Pull request merged**